Added buttonsOrder option

wenzhixin opened the pull request:
Fix #2135 

Example: https://live.bootstrap-table.com/code/wenzhixin/1191

ref: https://github.com/wenzhixin/bootstrap-table-examples/pull/387

UtechtDustin commented:
The buttons will broke, if you move the columns after paginationSwtich in the Foundation theme.
https://live.bootstrap-table.com/code/UtechtDustin/1204

![Bildschirmfoto vom 2019-11-19 09-37-22](https://user-images.githubusercontent.com/13292481/69130098-3650ac80-0ab0-11ea-8321-329fba4a1e50.png)

wenzhixin commented:
* Fixed foundation dropdown html error.
* Added `data-*` support, for example: 
`data-buttons-order="['columns', 'paginationSwitch', 'refresh', 'toggle', 'fullscreen']"`
`data-buttons-order="[columns, paginationSwitch, refresh, toggle, fullscreen]"`
`data-buttons-order="[columns,paginationSwitch,refresh,toggle,fullscreen]"`

wenzhixin commented:
![image](https://user-images.githubusercontent.com/2117018/69157216-99c5f300-0b1f-11ea-8b3b-138c97337560.png)

djhvscf commented on src/bootstrap-table.js:
```diff
@@ -503,78 +503,84 @@ class BootstrapTable {
+      fullscreen: `<button class="${this.constants.buttonsClass}" type="button" name="fullscreen"
         aria-label="Fullscreen" title="${o.formatFullscreen()}">
         ${o.showButtonIcons ? Utils.sprintf(this.constants.html.icon, o.iconsPrefix, o.icons.fullscreen) : '' }
         ${o.showButtonText ? o.formatFullscreen() : ''}
```
I know this is not part of this PR, but can we change the `o`variable name to something more meaningful like `opts`?

wenzhixin replied:
Done!
